index.js

```diff
@@ -19,22 +20,26 @@ function convertDate (d: Date): number {
     (d.getHours() << 11) | (d.getMinutes() << 5) | (d.getSeconds() >> 1)
 }
 
-// Filter hidden files/directories.
+// Filter hidden.
 function filterHidden (item: string) {
   const basename = path.basename(item)
   return basename === '.' || basename[0] !== '.'
 }
 
 // Filter directories.
-function filterDirectories (exclude: Array) {
-  return function (item: string) {
-    for (const dir of exclude) {
-      if (item.startsWith(dir)) {
-        return false
+const filterDirectories = function (opts: Object) {
```

Comment:
I would prefer to not bring in a new dependency, which will trigger audits downstream. I'd like it if you could rewrite this code without having to use `through2`.

Reply:
I've updated to avoid using `through2`.

There's one point I'd like to draw attention to: the path in the options is relative, whereas the path passed by `klaw` is absolute. Because of this I'm having to resort to check: `item.path.indexOf` to determine whether to exclude or not. This makes the option a exclude pattern, to be more accurate. Do you see any other issues with this approach?

Reply:
As an example, in our current tests:

adding a file: `'test/fixtures/f.txt', will cause test case `all files except hidden and ignored dir` to ignore the file also, because it's path contains the exclude path.

Reply:
> Do you see any other issues with this approach?

It looks good to me, and the tests vindicate you. 😺